pkg/udf/forward/forward.go

```diff
@@ -73,17 +73,7 @@ type InterStepDataForward struct {
 }
 
 // NewInterStepDataForward creates an inter-step forwarder.
-func NewInterStepDataForward(
-	vertexInstance *dfv1.VertexInstance,
-	fromStep isb.BufferReader,
-	toSteps map[string][]isb.BufferWriter,
-	fsd forwarder.ToWhichStepDecider,
-	applyUDF applier.MapApplier,
-	applyUDFStream applier.MapStreamApplier,
-	fetchWatermark fetch.Fetcher,
-	publishWatermark map[string]publish.Publisher,
-	idleManager wmb.IdleManager,
-	opts ...Option) (*InterStepDataForward, error) {
+func NewInterStepDataForward(vertexInstance *dfv1.VertexInstance, fromStep isb.BufferReader, toSteps map[string][]isb.BufferWriter, fsd forwarder.ToWhichStepDecider, applyUDF applier.MapApplier, applyUDFStream applier.MapStreamApplier, applyUDFBatchMap applier.BatchMapApplier, fetchWatermark fetch.Fetcher, publishWatermark map[string]publish.Publisher, idleManager wmb.IdleManager, opts ...Option) (*InterStepDataForward, error) {
```

Comment:
can you merge all the appliers into a struct? the list is arg list is exploding :)